File: examples/swift-server/Sources/AppwriteModels/LogList.swift

/// Logs List
public class LogList {

    /// List of logs.
    public let logs: [Log]

    init(
        logs: [Log]
    ) {
        self.logs = logs
    }

    public static func from(map: [String: Any]) -> LogList {
        return LogList(
            logs: (map["logs"] as! [[String: Any]]).map { Log.from(map: $0) }
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "logs": logs.map { $0.toMap() } as Any
        ]
    }
                                                                                                                                                                
}

File: examples/flutter/lib/src/models/log_list.dart
part of appwrite.models;

/// Logs List
class LogList {
    /// List of logs.
    final List<Log> logs;

    LogList({
        required this.logs,
    });

    factory LogList.fromMap(Map<String, dynamic> map) {
        return LogList(
            logs: List<Log>.from(map['logs'].map((p) => Log.fromMap(p))),
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "logs": logs.map((p) => p.toMap()),
        };
    }
}
